numpy/core/arrayprint.py

```diff
@@ -1218,19 +1228,19 @@ def format_float_positional(x, precision=None, unique=True,
                               sign=sign, pad_left=pad_left,
                               pad_right=pad_right, min_digits=min_digits)
 
-
 class IntegerFormat:
-    def __init__(self, data):
+    def __init__(self, data, sign=None):
         if data.size > 0:
             max_str_len = max(len(str(np.max(data))),
                               len(str(np.min(data))))
+            if sign in ['+', ' ']:
+                max_str_len += 1
         else:
             max_str_len = 0
-        self.format = '%{}d'.format(max_str_len)
+        self.format = lambda x: f"{x:{sign}{max_str_len}d}"
```

Comment:
I still think that `self.format = f"{{:{sign}{max_str_len}d}}"` makes more sense (also dropping the lambda).  Why interpolate `sign` and `max_str_len` potentially thousands of times?  (yes, a bit of a micro-optimization, do you prefer the lambda?).  (EDIT: no need for `x`, we will format positionally)